src/NUnitEngine/nunit.engine/Drivers/NUnit2DriverFactory.cs

```diff
@@ -57,9 +58,9 @@ public bool IsSupportedTestFramework(string assemblyName, Version version)
         /// <param name="assemblyName">The name of the test framework reference</param>
         /// <param name="version">The version of the test framework reference</param>
         /// <returns></returns>
-        public IFrameworkDriver GetDriver(AppDomain domain, string assemblyName, Version version)
+        public IFrameworkDriver GetDriver(AppDomain domain, AssemblyName reference)
         {
-            if (!IsSupportedTestFramework(assemblyName, version))
+            if (!IsSupportedTestFramework(reference))
                 throw new ArgumentException("Invalid framework name", "frameworkAssemblyName");
```

Comment:
Minor point since it was wrong before your changes, but the ArgumentException is using the wrong parameter name. I have this new branch local, so I will fix it. I wish we could start using nameof() here. I think it would work everywhere except CF.